numpy/core/src/umath/loops.c.src

```diff
@@ -1566,9 +1568,12 @@ NPY_NO_EXPORT void
     char * margs[] = {args[0], args[0], args[1]};
     npy_intp msteps[] = {steps[0], steps[0], steps[1]};
     if (!@isnan@ || !run_binary_simd_not_equal_@TYPE@(margs, dimensions, msteps)) {
-        UNARY_LOOP {
-            const @type@ in1 = *(@type@ *)ip1;
-            *((npy_bool *)op1) = @func@(in1) != 0;
+        if (!(@isfinite@ && run_isfinite_simd_@TYPE@(args, dimensions, steps)) &&
```

Comment:
It would be good to break these out into separate repeats, the expression in the if statement here is getting out of hand.